sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -99,21 +103,24 @@ export class IdentityClient extends ServiceClient {
       (refreshParams as any).client_secret = clientSecret;
     }
 
-    const webResource = this.createWebResource({
-      url: `${this.authorityHost}/${tenantId}/oauth2/v2.0/token`,
-      method: "POST",
-      disableJsonStringifyOnBody: true,
-      deserializationMapper: undefined,
-      body: qs.stringify(refreshParams),
-      headers: {
-        Accept: "application/json",
-        "Content-Type": "application/x-www-form-urlencoded"
-      },
-      abortSignal: options && options.abortSignal
-    });
-
     try {
-      return await this.sendTokenRequest(webResource, expiresOnParser);
+
+      const webResource = this.createWebResource({
+        url: `${this.authorityHost}/${tenantId}/oauth2/v2.0/token`,
+        method: "POST",
+        disableJsonStringifyOnBody: true,
+        deserializationMapper: undefined,
+        body: qs.stringify(refreshParams),
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/x-www-form-urlencoded"
+        },
+        spanOptions: newOptions.spanOptions,
+        abortSignal: options && options.abortSignal
+      });
+
+      const response = await this.sendTokenRequest(webResource, expiresOnParser);
+      return response;
```

Comment:
Should we set the status to CanonicalCode.Ok if we got the response?

Reply:
OK is the default. Spans that aren't given an explicit status are always OK

Reply:
Ah good to know! Guess I don’t need to set that in my PR then :-)